### DISCUSSION TITLE
I2S on STM32

### DISCUSSION THREAD

**Discussion by farukhalilagic** [zephyrproject-rtos/zephyr#58276] [Q&A]
I'm trying to enable I2S periperal on STM32F429ZI. My overlay file looks like this:

i2s3_sd_pc12:i2s3_sd_pc12{
pinmux = <STM32_PINMUX('C',12,AF6)>;
};

i2s3_ck_pc10:i2s3_ck_pc10{
pinmux = <STM32_PINMUX('C',10,AF6)>;
};

i2s3_mck_pc7:i2s3_mck_pc7{
pinmux = <STM32_PINMUX('C',7,AF6)>;
};

i2s3_ws_pa4:i2s3_ws_pa4{
pinmux = <STM32_PINMUX('A',4,AF6)>;
};
};

&dma1{
status = "okay";
};

&plli2s{
status = "okay";
mul-n=<100>;
div-r=<2>;
};

&i2s3{
mck-enabled;
clocks = < &rcc 0x40 0x8000 >, < &rcc  STM32_SRC_PLL_R I2S_SEL(0)>;
pinctrl-0 = <&i2s3_ws_pa4  &i2s3_mck_pc7 &i2s3_ck_pc10 &i2s3_sd_pc12>;
pinctrl-names = "default";
status="okay";
};

But I get error:
[00:00:00.000,000] <err> i2s_ll_stm32: Could not configure I2S domain clock
[00:00:00.000,000] <err> i2s_ll_stm32: i2s_stm32_initialize: clock enabling failed: -5[00:00:00.208,000] <err> i2s_ll_stm32: Failed call clock_control_get_rate(pclken[1])

Can anyone give an example of using I2S with STM32 microcontrollers ?

**Comment by chrisschir** [comment]
@farukhalilagic Since I have the same problem, have you found a solution?

**Comment by ndrs-pst** [comment]
Since the error occurred on the following lines, it would be helpful if you can share `zephyr.dts`.
Are you using a custom board or an in-tree board such as the `nucleo_f429zi`?

https://github.com/zephyrproject-rtos/zephyr/blob/7c64a8b1a7ad7d7330c27affcf0569713c922bcf/drivers/i2s/i2s_ll_stm32.c#L120-L129

**Reply by chrisschir** [reply]
I am using an U585 on the b_u585i_iot02a evaluation board. But since I got the same error as described above I thought it might be an error in the I2S configuration which both of us did wrong and not a device dependent error. The return value of clock_control_on is 134 (Unsupported value).
My I2S configuration looks like this:
```c
i2s2: i2s@40015800 {
    compatible = "st,stm32-i2s";
    reg = <0x40015800 0x400>;
    interrupts = <38 IRQ_TYPE_LEVEL_HIGH>;
    interrupt-parent = <&intc>;
    #address-cells = <1>;
    #size-cells = <0>;
    clocks = <&rcc STM32_CLOCK_BUS_APB2 0x00400000>,
           <&rcc STM32_SRC_PLL2_R I2S_SEL(0)>;
    dmas = <&gpdma1 0 38 0x09680>,
            <&gpdma1 1 39 0x09640>;
    dma-names = "tx", "rx";
    status = "disabled";
  };
``` 
and
```c
&i2s2 {
  status = "okay";
  pinctrl-0 = <&i2s2_ck_pb13 &i2s2_mck_pb14 &i2s2_sdi_pg12 &i2s2_sdo_pc12 &i2s2_ws_pg10>;
  pinctrl-names = "default";
  mck-enabled;
};
```

**Comment by ajarmouni-st** [comment]
> Can anyone give an example of using I2S with STM32 microcontrollers ?

See https://github.com/zephyrproject-rtos/zephyr/blob/65a80d5bc286b7c3e0abe7d059eda2d9de881b1c/boards/st/nucleo_f411re/nucleo_f411re.dts#L108-L111

**Comment by FRASTM** [comment]
maybe the clock_control definition stm32f4_clock.h/stm32f427_clock.h are not really suitable for the stm32f429 (RM0090)
 The configuration register for the PLL I2S is RCC PLLI2SCFGR at offset 0x84
As the plli2s is enabled, I would expect to have PLLI2S enabled and I2S clock source by the PLLI2S without domain clock, in that case

Can you see with this branch https://github.com/FRASTM/zephyr/tree/stm32f4_i2s if it helps ?
